src/helpers/credential.rs

```diff
@@ -269,6 +279,138 @@ impl<'a> Credential<'a> {
+            .get_issuer_public_key(&subject_did, "#bbs-key-1")
+            .await?;
+        let issuer_public_key_copy= issuer_public_key.clone();
+        let use_valid_until= if exp_date.is_some() { true } else { false };
+        let exp_date_str = exp_date.unwrap_or("");
+
+        // need to add chrono crate
+        // let date_now = SystemTime::now();
+        // let datetime: DateTime<Utc> = date_now.into();
+        // let date_now_str = datetime.format("%d/%m/%Y %T");
+
+        // create credential offer
+        let unsigned_credential = UnsignedBbsCredential {
+            context: vec![
+                "https://www.w3.org/2018/credentials/v1".to_string(),
+                "https://schema.org/".to_string(),
+                "https://w3id.org/vc-revocation-list-2020/v1".to_string(),
+            ],
+            id: "uuid:834ca9da-9f09-4359-8264-c890de13cdc8".to_string(),
+            r#type: vec!["VerifiableCredential".to_string()],
+            issuer: subject_did,
+            valid_until: if use_valid_until {
+                Some(exp_date_str.to_string())
+            } else {
+                None
+            },
+            issuance_date: "2023-02-22T00:00:00.000Z".to_string(), //date_now_str.to_string(),
+            credential_subject: credential_subject_copy,
+            credential_schema: CredentialSchemaReference {
+                id: schema.id,
+                r#type: schema.r#type,
+            },
+            credential_status: CredentialStatus {
+                id: "did:evan:zkp:placeholder_status#0".to_string(),
+                r#type: "RevocationList2020Status".to_string(),
+                revocation_list_index: "0".to_string(),
+                revocation_list_credential: "did:evan:zkp:placeholder_status".to_string(),
+            },
+        };
+        let unsigned_credential_str = serde_json::to_string(&unsigned_credential)?;
+        let nquads = convert_to_nquads(&unsigned_credential_str).await?;
+
+        let payload = OfferCredentialPayload {
+            issuer: subject_did_str_copy.to_string(),
+            subject: Option::from(subject_did_copy),
+            nquad_count: nquads.len(),
+        };
+        let offer_str = self
+            .vade_evan
+            .vc_zkp_create_credential_offer(
+                EVAN_METHOD,
+                TYPE_OPTIONS,
+                &serde_json::to_string(&payload)?,
+            )
+            .await
+            .map_err(|err| CredentialError::VadeEvanError(err.to_string()))?;
+
+        // Create credential request
+        let credential_values_str = serde_json::to_string(&credential_subject.data)?;
+        let request_str = self
+            .create_credential_request(
+                serde_json::to_string(&(issuer_public_key.as_str()))?.as_str(),
+                serde_json::to_string(bbs_secret)?.as_str(),
+                credential_values_str.as_str(),
+                offer_str.as_str(),
+                schema_did,
```

Comment:
```suggestion
                &issuer_public_key,
                bbs_secret,
                &credential_values_str,
                &offer_str,
                schema_did,
```

Reply:
Here we have an interesting culprit - f.e. when I passed bbs_secret as r#""blah-blah-blah""# (double double quotes) from the test, I got quotes in the string, and it was legal JSON representation, but the bbs_secret will be passed in single pair of double quotes in the real life, and credential_request() will fail (it did, I tried), complaining about invalid JSON.  So, for issuer_public_key and bbs_secret I had to use serde_json::to_string.

This is the error I have after changing line 361 to suggested:
Error: internal VadeEvan call failed; vade call failed with: could not run vc_zkp_request_credential for "did:evan"; expected value at line 5 column 33 when parsing payload {
                "credentialOffering": {"issuer":"did:evan:EiAJqjQKlTF4lsgUKL1Tvmvgm3ECXtyVA08Optz8k_W6Lg","subject":"did:evan:EiAJqjQKlTF4lsgUKL1Tvmvgm3ECXtyVA08Optz8k_W6Lg","nonce":"Ya/pTcKnFAC7BFddZoglGWQv6nEB0K+v2dsXv/RF6VI=","credentialMessageCount":13},
                "masterSecret": "OASkVMA8q6b3qJuabvgaN9K1mKoqptCv4SCNvRmnWuI=",
                "credentialValues": {"email":"value@x.com"},
                "issuerPubKey": lxx0s6mk6CgRD+2e+S4npgKUPHSCxdWa3eSvxKOzMv3XSTKYxArKQ3Ep3Uh+cTy2EVMzsR2GwJ4benLnKPzxIOxwLoAvypdoBffH4uTUvTME/CouswxSaBQEEdxjy8Hy,
                "credentialSchema": {"id":"did:evan:EiACv4q04NPkNRXQzQHOEMa3r1p_uINgX75VYP2gaK5ADw","type":"EvanVCSchema","name":"equs.contact.email","author":"did:evan:EiDkld4Y414kqnn_I5yi9Zfwic8ToumwIpCaJKwVgTz31w","createdAt":"2022-08-22T13:26:04.000Z","description":"Contains the equs verified email.","properties":{"email":{"type":"string"}},"required":["email"],"additionalProperties":false,"proof":{"type":"EcdsaPublicKeySecp256k1","created":"2022-08-22T13:26:04.000Z","proofPurpose":"assertionMethod","verificationMethod":"did:evan:EiDkld4Y414kqnn_I5yi9Zfwic8ToumwIpCaJKwVgTz31w#key1","jws":"eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NkstUiJ9.eyJpYXQiOiIyMDIyLTA4LTIyVDEzOjI2OjA0LjAwMFoiLCJkb2MiOnsiaWQiOiJkaWQ6ZXZhbjpFaUFDdjRxMDROUGtOUlhRelFIT0VNYTNyMXBfdUlOZ1g3NVZZUDJnYUs1QUR3IiwidHlwZSI6IkV2YW5WQ1NjaGVtYSIsIm5hbWUiOiJlcXVzLmNvbnRhY3QuZW1haWwiLCJhdXRob3IiOiJkaWQ6ZXZhbjpFaURrbGQ0WTQxNGtxbm5fSTV5aTlaZndpYzhUb3Vtd0lwQ2FKS3dWZ1R6MzF3IiwiY3JlYXRlZEF0IjoiMjAyMi0wOC0yMlQxMzoyNjowNC4wMDBaIiwiZGVzY3JpcHRpb24iOiJDb250YWlucyB0aGUgZXF1cyB2ZXJpZmllZCBlbWFpbC4iLCJwcm9wZXJ0aWVzIjp7ImVtYWlsIjp7InR5cGUiOiJzdHJpbmcifX0sInJlcXVpcmVkIjpbImVtYWlsIl0sImFkZGl0aW9uYWxQcm9wZXJ0aWVzIjpmYWxzZX0sImlzcyI6ImRpZDpldmFuOkVpRGtsZDRZNDE0a3Fubl9JNXlpOVpmd2ljOFRvdW13SXBDYUpLd1ZnVHozMXcifQ.SJJHIxwrkwQvsPzsvIZ1Ws5siJSBEQaroegRtTJUzut1MiZaF-QEQQcEMoXKEcndc-60X-DYXmC9qBOXpirX2Bw"}}
            }

Reply:
When we use serde, issuePubKey is "lxx0s...", not lxx0s...

Reply:
Although, it is ok for &credential_values_str and &offer_str, as they are structs and stringified in curly braces.